### PR TITLE
Add dmic and pdm classes to topology2

### PR DESCRIPTION
**Pull request by juimonen**

### CODE REVIEW

File: tools/topology/topology2/include/dais/pdm_config.conf [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+	DefineAttribute.clk_edge {
+		token_ref	"sof_tkn_dmic_pdm.short"
+	}
+
+	DefineAttribute.skew {
+		token_ref	"sof_tkn_dmic_pdm.short"
+	}
+
+	attributes {
+		constructor [
```
Comment (aiChaoSONG) [inline tools/topology/topology2/include/dais/pdm_config.conf:34]: please add `!` before `constructor` and `mandatory`, and make them `!constructor` and `!mandatory`, for the reason, please check https://github.com/alsa-project/alsa-lib/issues/151

Comment (juimonen) [reply tools/topology/topology2/include/dais/pdm_config.conf]: ok added now

**Comment by ranj063** [comment]
@juimonen could you please check if the class definition is the same as the one in the wip branch? If so, we should be good to go

**Comment by aiChaoSONG** [comment]
@ranj063 helped to check, Jask added some more comments and add default value for `default_hw_config_id`, this should be good.

**Review by aiChaoSONG — APPROVED** [review]

File: tools/topology/topology2/include/dais/dmic.conf [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+	DefineAttribute.duty_min {
+		token_ref	"sof_tkn_dmic.word"
+	}
+
+	DefineAttribute.duty_max {
+		token_ref	"sof_tkn_dmic.word"
+	}
+
+	DefineAttribute.num_pdm_active {
+		token_ref	"sof_tkn_dmic.word"
```
Comment (aiChaoSONG) [inline tools/topology/topology2/include/dais/dmic.conf:73]: @juimonen  Just find that some used tokens are not defined, `sof_tkn_dmic`,  `sof_tkn_dmic_pdm`. please define them in common/tokens.conf

Comment (juimonen) [reply tools/topology/topology2/include/dais/dmic.conf:73]: ok.

**Review by singalsu — APPROVED** [review]

File: tools/topology/topology2/include/common/tokens.conf [hunk 3]
```diff
@@ -73,6 +73,31 @@ Object.Base.VendorToken {
...
+	"sof_tkn_intel_dmic_pdm" {
+		ctrl_id		700
+		mic_a_enable	701
+		mic_b_enable	702
+		polarity_a	703
+		polarity_b	704
+		clk_edge	705
+		skew		706
+
+	}
```
[the remark this answers is not in the document]
Comment (juimonen) [reply tools/topology/topology2/include/common/tokens.conf:99]: fir parameters are visible in the second commit:
                fir_type                 "kaiser-linear-phase"
                fir_length               127
                cic_decimate             25
                fir_decimate             2
                fir_passband_end         20000
                fir_stopband_start       24000
                fir_passpand_ripple      10
                fir_stopband_ripple      -95000

**Pull request merged**